src/Event/AbstractStorageEventListener.php

```diff
@@ -137,16 +139,9 @@ public function buildFilename($table, $entity) {
  * @return string
  */
 	public function buildPath($table, $entity) {
-		$path = '';
-		if ($this->_config['tableFolder']) {
-			$path .= $table->table() . DS;
-		}
-		if ($this->_config['randomPath'] === true) {
-			$path .= StorageUtils::randomPath($entity[$table->primaryKey()]);
-		}
-		if ($this->_config['uuidFolder'] === true) {
-			$path .= $this->stripDashes($entity[$table->primaryKey()]) . DS;
-		}
+		$pathBuilder = $this->createPathBuilder($entity['adapter']);
```

Comment:
This is **not** going to work because the "adapter" (which should have been named config in the past) is a configuration name that can be anything. It was never thought to be used as a convention. Also the idea is in fact that the path builder *doesn't* have to match the adapter name to allow people to customize the path building. I won't accept this part of the changes.